### ISSUE TITLE
Temporary Suppressions generator seems to count occurrences wrong

### ISSUE THREAD

**Issue by JKrag** [atc-net/atc-coding-rules-updater#47]
When running the updater with --useTemporarySuppresions, it adds lines to .editorconfig for each violating rule. The comment generated for each violation includes the number of occurrences found. This is quite useful when triaging which problems are easy to attack and clean up.
I have however seen a few examples where the updater vastly underestimates the number of violations. E.g. my .editorconfig after running mentioned "# occurrences 13 ..." for SA1503, but when I commented out the suppression, and rana  build, there were way over 50 errors.

I am running on a solution with 5 projects, so maybe the updater is just counting the occurrences in the first project it finds, generating the exclusion, and then somehow not adding to the number for later projects analysed? (pure guess)  

This is not a critical issue by any means, but worth looking at if this part of the code is revisited for other reasons.
If the generator bothers to mention the number of occurrences, it does look better if it is actually correct 😄

**Comment by davidkallesen** [comment]
Good idea, and cheap — the version was already available internally for `--version`. Added:

```text
##########################################
# ATC temporary suppressions
# generated @ Wednesday, July 29, 2026 3:06:40 PM
# generated by: atc-coding-rules-updater v2.0.264
# Please fix all generated temporary suppressions
...
```

Two notes on the details:

- It is written in **both** places the header is produced — the block appended to `.editorconfig` and
  the standalone file written when `--temporarySuppressionPath` is set — so the two cannot drift apart.
- Trimmed to three parts rather than the four your example implied. Internally the version is
  `2.0.264.32990`, where the last component is the git height added by Nerdbank.GitVersioning. Three
  parts matches the package version you would actually install.

While in that code, one adjacent defect your issue indirectly surfaced also got fixed: the
`.editorconfig` timestamp was formatted with the machine's current culture, while the standalone file
used English explicitly. On a non-English machine the committed `.editorconfig` picked up localised
month and day names, so the same command produced a different file depending on who ran it. Both sites
now use the same culture.

Fixed in PR #188
